cli/pkg/config/happy_version_lock.go

```diff
@@ -0,0 +1,92 @@
+package config
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+)
+
+type HappyVersionLockFile struct {
+	HappyVersion string
+	Path         string `json:"-"`
+}
+
+func NewHappyVersionLockFile(projectRoot string, requiredVersion string) (*HappyVersionLockFile, error) {
+
+	if projectRoot == "" {
+		return nil, errors.New("No projectRoot specified")
+	}
+
+	if requiredVersion == "" {
+		return nil, errors.New("No requiredVersion specified")
+	}
```

Comment:
```suggestion
```

To avoid too much error checking all over the place, I usually only return errors when I really need to fail (like a network request has gone bad or something). For constructors like this where the parameters are required of the function, I think that is enough to the developer that they can't be empty. Remove the error from the return

Reply:
Ok, removed